packages/next/src/withTRPC.tsx

```diff
@@ -205,11 +204,13 @@ export function withTRPC<
           trpc: trpcProp,
         };
 
+        const reactDomServer = await import('react-dom/server');
+
         // Run the prepass step on AppTree. This will run all trpc queries on the server.
         // multiple prepass ensures that we can do batching on the server
         while (true) {
           // render full tree
-          await ssrPrepass(createElement(AppTree, prepassProps as any));
```

Comment:
seems like this was never needed in the first place? 

we might just want to do this on `main` if this works well

Reply:
Actually, it might be needed in some setups.
We need some render function here, and if we use `renderToString` all the tests pass, but if we add a test with a suspended component (which suspension is not based on `queryClient`) that has a `useQuery` inside - it might fail, we'll have no queries running on first render, but there will be after suspension resolution.
I'm not sure if that weird case should be supported or not, but the expected functions might break silently with that change.

Let me know if you're in to make this change on main branch.

Reply:
SSR wasn't working together with suspense before so this isn't regression at least, I think we even noted that in the docs